dartsim/src/SDFFeatures_TEST.cc

```diff
@@ -559,8 +598,13 @@ TEST_P(SDFFeatures_TEST, WorldWithNestedModel)
   auto nestedModel = parentModel->GetNestedModel("nested_model");
   ASSERT_NE(nullptr, nestedModel);
 
+  /// \todo(srmainwaring) this is failing. DART associates the nested joint
+  /// with the skeleton of the top level model when the nested model is
+  /// joined to the parent model, but Gazebo should not find grandchild
+  /// joints when querying a parent model.
   auto nestedJoint = parentModel->GetJoint("nested_joint");
-  EXPECT_NE(nullptr, nestedJoint);
+  // EXPECT_NE(nullptr, nestedJoint);
+  EXPECT_EQ(nullptr, nestedJoint);
```

Comment:
I agree with your analysis and the change you made here. We should also add

```
EXPECT_NE(nullptr, nestedModel->GetJoint("nested_joint"));
```

Reply:
Updated in 539e8be6.

Reply:
@azeey thanks for the feedback. I've made the changes in a series of commits in case we want to revert any of them.

I've left some debug helpers in Base.h with \todo removes against them. I found them very useful for tracing what was going on in the various maps. Let me know if you'd like them removed or left (in which case I'll update the comment).

Also, in JointFeatures.cc I've left three \todo's about consolidating some common code (generating fully scoped names for joints). I was going to add a method in an anonymously namespace but the function to generate identities are protected (so I can't pass the this pointer to a non-class function to do the work). Can leave as is or consolidate in a member function if you think that would be better.

Reply:
Since these classes aren't directly exposed to the user, i.e, the plugin implementation is an internal detail, we don't have a lot of constraints as to what should be a free function vs a member function, so please go ahead and consolidate the common code in a member function.

Reply:
If you're ok with leaving helper functions, then I'll remove the `\todo` comments from

https://github.com/gazebosim/gz-physics/blob/b8b6842d3ed990e849c2379f118584c053ee2db2/dartsim/src/Base.hh#L644-L668

etc. and leave them in place as they are very useful for debugging (I find it's difficult to inspect the maps directly in gdb / lldb). I'll add a helper for the commonly used code creating the fully scoped name and place it where it can be accessed as required.